src/app/booking/OOSPlannerSection.js

```diff
@@ -183,23 +183,27 @@ const StyledTableCell = styled(TableCell)(({ theme }) => ({
 }));
 
 function FamilyBookings() {
-  const today = dayjs();
+  const { selectedDate, setSelectedDate } = useContext(SelectedDateContext);
 
-  let currentWeekFirstDayDate;
-  if (today.weekday() === 0 || today.weekday() === 6) {
-    currentWeekFirstDayDate = today.startOf("week").add(1, "day");
-  } else {
-    currentWeekFirstDayDate = today.startOf("week").add(1, "day");
-  }
-  const [weekFirstDayDate, setWeekFirstDayDate] = useState(currentWeekFirstDayDate);
-  const weekDates = Array.from({ length: 5 }, (_, i) => weekFirstDayDate.add(i, "day"));
+  const getStartOfWeek = date => {
+    let day = date.startOf('week').add(1, 'day');
+    if (date.weekday() === 6) {
```

Comment:
Also, you have 2 ifs for Saturday and Sunday. But it could be 1 if: you can just add 7 days and use startOf('week')